### ISSUE TITLE
BraintreeCancelListener#onCancel invoked twice

### ISSUE THREAD

**Issue by weefbellington** [braintree/braintree_android#112]
I've identified what appears to be a bug in the Braintree Android SDK. It looks like BraintreeFragment#postCancelCallback is called multiple times after the PayPal activity is canceled by the user.

This means that any `BraintreeCancelListener` that is registered with the BraintreeFragment will receive the `BraintreeCancelListener#onCancel` callback twice.

If `BraintreeFragment#removeListener` is is invoked as part of the first onCancel callback, the second cancel event is queued. When `BraintreeFragment#addListener` is called, the queued event is flushed and `BraintreeCancelListener#onCancel` is invoked immediately.

Because of this issue, a user may successfully authorize a PayPal future payment, but an app will track it as a cancellation instead. The PayPal payment nonce is never uploaded to our server.

## STEPS TO REPRODUCE

1. ensure paypal app is not installed on the phone
2. from a client app activity, call PayPal#authorizeAccount(braintreeFragment), which triggers the browser switcher
3. press the back button to return to the client app activity

## NOTES

- postCancelCallback is called first here: https://github.com/braintree/braintree_android/blob/4e5c024e21633a87552d52be8f973a80760a11c0/Braintree/src/main/java/com/braintreepayments/api/PayPal.java#L399
- postCancelCallback is called a second time here:
https://github.com/braintree/braintree_android/blob/ae956caf720335f54ed4cb7601d940ba8841ed75/Braintree/src/main/java/com/braintreepayments/api/BraintreeFragment.java#L296
- the home button has the same behavior as the back button

**Comment by weefbellington** [comment]
I should mention the Braintree SDK version that I'm using:
```groovy
compile 'com.braintreepayments.api:braintree:2.3.3'
compile 'com.braintreepayments.api:data-collector:2.3.3'
```

**Comment by sdcoffey** [comment]
Hey @weefbellington,

Thanks for bringing this to our attention! Pushing a fix for this now. I'll close this issue when it's merged and released.

**Comment by lkorth** [comment]
@weefbellington thanks for the report. We've pushed a fix to `master` and will be releasing it shortly.

**Comment by lkorth** [comment]
This was released in `2.3.6`.
